lib/Language/Souffle/Interpreted.hs

```diff
@@ -188,6 +190,18 @@ cleanup (Handle ref) = liftIO $ do
   handle <- readIORef ref
   traverse_ removeDirectoryRecursive [factPath handle, outputPath handle, basePath handle]
 
+locateSouffle :: IO (Maybe FilePath)
+locateSouffle = do
+  let locateCmd = (shell "whereis souffle")
+                    { std_out = CreatePipe
+                    }
+  (_, Just hout, _, locateCmdHandle) <- createProcess locateCmd
+  waitForProcess locateCmdHandle >>= \case
+    ExitFailure _ -> pure Nothing
+    ExitSuccess   -> do
+      fmap words (hGetContents hout) >>= \case
+        (_souffle : souffleBin : _) -> pure (souffleBin `deepseq` Just souffleBin)
```

Comment:
Not sure if this deepseq is needed? Will play around with it.

Reply:
Ack